views/providers.py

```diff
@@ -179,13 +179,17 @@ def add_provider_if_not_present(self):
     def does_provider_exist(self):
         self.web_session.logger.info("Checking if provider exists")
 
-        # For performance reasons, check if the provider is present via DB
-        providers = db(self.web_session).get_providers()
-        provider = ui_utils(self.web_session).find_row_in_list(providers, 'name', self.web_session.HAWKULAR_PROVIDER_NAME)
-        if provider:
-            return True
-        else:
+        # navigate_to_providers
+
+        NavigationTree(self.web_session).navigate_to_middleware_providers_view()
+
+        if ui_utils(self.web_session).isElementPresent(By.XPATH, "//span[contains(.,'(Item 0 of 0)')]"):
+            self.web_session.logger.info("Middleware Provider does not exist.")
             return False
```

Comment:
Is there a reason why the performance enactment was reverted back to previous method of checking for existing Provider?